### DISCUSSION TITLE
Updated OneDrive client to 2.5. Updated OneDriveGUI to 1.1 . Problem on 1.1 GUI startup

### DISCUSSION THREAD

**Discussion by cliverlong-1961** [bpozdena/OneDriveGUI#186] [Q&A] [answered]
I am running Ubuntu 22.04 and _was_ successfully running the OneDrive client with OneDriveGUI 

I saw a prompt to update OneDrive client so I updated to 2.5

See: 

> 
> onedrive --version
> onedrive v2.5.0-1+np2+2.1
> 
>

When I ran 

> onedrive --display-config

I received following response

> D-Bus message bus daemon is available; GUI notifications are now enabled
> Reading configuration file: /home/clive/.config/onedrive/config
> Invalid value for key in config file - using default value: log_dir
> 
> Invalid value for key in config file: drive_id

The maintainer of OneDrive client advised me to update OneDriveGUI and there is a note to that effect [here](https://github.com/bpozdena/OneDriveGUI/tree/main)

> Warning
> 
> OneDriveGUI v1.1.0 is only compatible with OneDrive client v2.5.0 . There is no backward compatibility with older OneDrive client versions. Please use OneDriveGUI v1.0.3 for OneDrive client v2.4.x .

So I followed the installation instructions for OneDriveGUI [here](https://github.com/bpozdena/OneDriveGUI/tree/main).

When I boot Ubuntu I see the OneDrive cloud in the top (status?) line of the display. When I make changes to the OneDrive folder on Linux or any device connected to the OneDrive folder, this is not synchronised across all devices. 

I stop the OneDrive GUI by right-clicking on the cloud icon then issue following commands from within OneDriveGUI folder on Ubuntu

> cd src
> cd OneDriveGUI
> python3 OneDriveGUI.py

and get following error messages

> Traceback (most recent call last):
>   File "/home/clive/OneDriveGUI/src/OneDriveGUI.py", line 65, in <module>
>     from ui.ui_login import Ui_LoginWindow
>   File "/home/clive/OneDriveGUI/src/ui/ui_login.py", line 18, in <module>
>     from PySide6.QtWebEngineWidgets import QWebEngineView
> ImportError: libshiboken6.abi3.so.6.2: cannot open shared object file: No such file or directory
> 

I have searched for the ImportError message but haven't been able to work out how that relates to my problem starting OneDriveGUI.

Any help to get OneDrive and OneDriveGUI working on my Ubunut system is appreciated.

**Comment by bpozdena** [comment]
Based on the traceback, you are running some old version of the GUI code for some reason. This was removed months ago. Just clone the latest code from github to fix the issue. 

Or better yet, simply use the [AppImage](https://github.com/bpozdena/OneDriveGUI/releases/tag/v1.1.0), which contains all necessary Python and QT dependencies.

**Reply by cliverlong-1961** [reply]
Hello,

Thank-you for you very prompt reply.

I am running a synchronisation of my onedrive folder very the command-line using 

> onedrive -s

as described[ here](https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#authorise-the-application-with-your-microsoft-onedrive-account)

That is running.

When I installed OneDriveGUI from the AppImage, I had the AppImage on my desktop and it created a OneDriveGUI folder with sub-folders : docs, scripts, src and other files.

I renamed the existing OneDriveGUI folder in my home directory to OneDriveGUI_old and copied/moved he desktop version of the OneDriveGUI folder into my home directory.

I navigated to the new ~/OneDriveGUI folder

executed

> cd src

> python3 OneDriveGUI.py 


and received messages

> 
> Traceback (most recent call last):
>   File "/home/clive/OneDriveGUI/src/OneDriveGUI.py", line 3235, in <module>
>     global_config = create_global_config()
>   File "/home/clive/OneDriveGUI/src/OneDriveGUI.py", line 2955, in create_global_config
>     _od_config = read_config(profile_config_file)
>   File "/home/clive/OneDriveGUI/src/OneDriveGUI.py", line 2866, in read_config
>     with open(config_file, "r") as f:
> FileNotFoundError: [Errno 2] No such file or directory: '/home/clive/.config/onedrive/accounts/cliverlong/config'

**Reply by cliverlong-1961** [reply]
I have run a OneDrive resync from command line using

> onedrive --sync --resync

And I recognise the files and folders being downloaded from OneDrive. The initial messages displayed are:

> D-Bus message bus daemon is available; GUI notifications are now enabled
> 
> The usage of --resync will delete your local 'onedrive' client state, thus no record of your current 'sync status' will exist.
> This has the potential to overwrite local versions of files with perhaps older versions of documents downloaded from OneDrive, resulting in local data loss.
> If in doubt, backup your local data before using --resync
> 
> Are you sure you wish to proceed with --resync? [Y/N] Y
> 
> Deleting the saved application sync status ...
> Using IPv4 and IPv6 (if configured) for all network operations
> Attempting to contact Microsoft OneDrive Login Service
> Successfully reached Microsoft OneDrive Login Service
> Configuring Global Azure AD Endpoints
> Fetching items from the OneDrive API for Drive ID: eaa39626be95a5df ................................................... 
> Processing 21751 applicable changes and items received from Microsoft OneDrive ...

**Reply by cliverlong-1961** [reply]
I have downloaded the AppImage again, made it executable and rerun it. I get similar error messages again. 

> clive@MacPro-Linux: ~ /Downloads$ chmod +x ./OneDriveGUI-1.1.0-x86_64.AppImage 
> clive@MacPro-Linux: ~ /Downloads$ ./OneDriveGUI-1.1.0-x86_64.AppImage 
> Traceback (most recent call last):
>   File "/tmp/.mount_OneDri5Hucjh/usr/src/OneDriveGUI.py", line 3235, in <module>
>     global_config = create_global_config()
>   File "/tmp/.mount_OneDri5Hucjh/usr/src/OneDriveGUI.py", line 2955, in create_global_config
>     _od_config = read_config(profile_config_file)
>   File "/tmp/.mount_OneDri5Hucjh/usr/src/OneDriveGUI.py", line 2866, in read_config
>     with open(config_file, "r") as f:
> FileNotFoundError: [Errno 2] No such file or directory: '/home/clive/.config/onedrive/accounts/cliverlong/config'
>

**Reply by bpozdena** [reply, marked as answer]
It clearly says your config file `/home/clive/.config/onedrive/accounts/cliverlong/config` does not exist. 

Did you check if it does? If so, share the output of 
`cat /home/clive/.config/onedrive/accounts/cliverlong/config`.


Based on your previous command, the config is in the default location `/home/clive/.config/onedrive/config`. If so, just delete the directory `/home/clive/.config/onedrive-gui/` , start the GUI again and import your profile config properly.

**Reply by cliverlong-1961** [reply]
I had resynched my OneDrive fodler overnight by issuing a --resync --sync command line to the OneDrive client. All my OneDrive folder contents downloaded except for a few files.

`/home/clive/.config/onedrive/config` did not exist but `/home/clive/.config/onedrive-gui/` did exist. This folder contained 

I deleted `/home/clive/.config/onedrive-gui/` as you suggested, reran `Python3 OneDriveGUI.py` which started up and ran through the authorization with my OneDrive personal account.

Following was created

> clive@MacPro-Linux:~/.config/onedrive/accounts/crlong_personal$ ls
> config         items.sqlite3      items.sqlite3-wal  sync_list
> config_backup  items.sqlite3-shm  refresh_token

After successfully starting and configuring OneDriveGUI, I tested a few folder creations and deletes on the linux system and another device with OneDrive client. First few folders successfully synchronized on creation or deletion on either system.

This is very good progress.

However, synchronization seemed to stop. I started adjusting some of the settings through OneDriveGUI and I currently have:

> clive@MacPro-Linux:~/.config/onedrive/accounts/crlong_personal$ cat config
> sync_dir = "/home/clive/OneDrive"
> monitor_interval = "30"
> enable_logging = "true"

I need to look at logs on both OneDriveGUI and OneDrive clients to see if there is something in my setup/configuration that i not detecting and triggering file/folder synchronization.

But overall, with your detailed help, very good progress.

Thanks
